src/main/java/fi/crosskey/card/vehicle/simulator/models/Helicopter.java

```diff
@@ -0,0 +1,78 @@
+package fi.crosskey.card.vehicle.simulator.models;
+
+import javax.persistence.*;
+
+import static fi.crosskey.card.vehicle.simulator.models.HelicopterState.*;
+
+@Entity
+public class Helicopter {
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    private long id;
+    private String name;
+    @Enumerated(EnumType.STRING)
+    private HelicopterState currentState;
+
+
+    public Helicopter(){
+        stateChange(LANDED);
+    }
+
+    public Helicopter(final String name, final HelicopterState state){
+        this.name = name;
+        stateChange(state);
+    }
+
+    public Helicopter(final String name){
+        this.name = name;
+        stateChange(LANDED);
+    }
+
+    public Helicopter(final HelicopterState state){
+        stateChange(state);
+    }
+
+
+    public void setCurrentState(final HelicopterState currentState) throws VehicleStateException {
+        stateValidator(currentState);
+
+    }
+
+    private void stateValidator(HelicopterState state) throws VehicleStateException {
```

Comment:
Naming: in object oriented programming, we tend to name methods using verbs and classes using nouns. Methods are actions, classes (and objects) are things.

`stateValidator` is a thing for me, not an action.

What's a more appropriate name for this method?